ezyfox-server-core/src/main/java/com/tvd12/ezyfoxserver/api/EzyAbstractResponseApi.java

```diff
@@ -53,26 +85,106 @@ protected final void secureResponse(
+                        transportType,
+                        bytes
+                    );
+                    sendPacketNow(session, packet);
+                } catch (Throwable e) {
+                    if (packet != null) {
+                        packet.release();
+                    }
+                    logger.info("response data now to session: {} failed", session, e);
+                }
             }
         } else {
             for (EzySession session : recipients) {
-                byte[] bytes = encryptMessageContent(messageContent, session.getSessionKey());
-                session.send(createPacket(bytes, pack));
+                EzyPacket packet = null;
+                try {
+                    byte[] bytes = encryptMessageContent(messageContent, session.getSessionKey());
+                    packet = createPacket(
+                        session,
+                        transportType,
+                        bytes
+                    );
+                    sendPacket(session, packet);
+                } catch (Throwable e) {
+                    if (packet != null) {
+                        packet.release();
+                    }
+                    logger.info("response data to session: {} failed", session, e);
+                }
             }
         }
     }
 
-    protected EzySimplePacket createPacket(Object bytes, EzyPackage pack) {
+    private EzyPacket createPacket(
+        EzySession session,
+        EzyConstant transportType,
+        Object bytes
+    ) {
+        EzyConstant actualTransportType = transportType;
+        if (actualTransportType == EzyTransportType.UDP_OR_TCP) {
+            actualTransportType = session.getDatagramChannelPool() != null
+                ? EzyTransportType.UDP
+                : EzyTransportType.TCP;
+        }
+        return createPacket(actualTransportType, bytes);
+    }
+
+    protected EzySimplePacket createPacket(
+        EzyConstant transportType,
+        Object bytes
+    ) {
         EzySimplePacket packet = new EzySimplePacket();
-        packet.setTransportType(pack.getTransportType());
+        packet.setTransportType(transportType);
         packet.setData(bytes);
         return packet;
     }
 
+    private void sendPacket(
+        EzySession session,
+        EzyPacket packet
+    ) throws Exception {
+        if (packet.getTransportType() == EzyTransportType.UDP) {
+            session.send(packet);
+        } else {
+            sendTcpPacket(session, packet);
+        }
+    }
+
+    protected void sendTcpPacket(
+        EzySession session,
+        EzyPacket packet
+    ) throws Exception {
+        session.send(packet);
+    }
+
+    private void sendPacketNow(
+        EzySession session,
+        EzyPacket packet
+    ) throws Exception {
+        if (packet.getTransportType() == EzyTransportType.UDP) {
+            session.sendNow(packet);
+        } else {
+            sendTcpPacketNow(session, packet);
+        }
```

Comment:
ditto

Reply:
`sendTcpPacketNow` is overridden by `EzySecureSocketResponseApi`